modules/creating-manifest-file-multiple-interfaces-br-ex.adoc

```diff
@@ -0,0 +1,213 @@
+
+. Define each interface definition in a machine configuration manifest file:
++
+.Example of a machine configuration file with definitions added for `bond1`, `eno1`, and `en02`
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 12-${ROLE}-sec-bridge-cni
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents:
+          source: data:;base64,<base-64-encoded-contents-for-bond1.conf>
+        path: /etc/NetworkManager/system-connections/bond1.nmconnection
+        filesystem: root
+        mode: 0600
+      - contents:
+          source: data:;base64,<base-64-encoded-contents-for-eno1.conf>
+        path: /etc/NetworkManager/system-connections/eno1.nmconnection
+        filesystem: root
+        mode: 0600
+      - contents:
+          source: data:;base64,<base-64-encoded-contents-for-eno2.conf>
+        path: /etc/NetworkManager/system-connections/eno2.nmconnection
+        filesystem: root
+        mode: 0600
+# ...
+----
+
+. Create a machine configuration manifest file for configuring the network plugin by entering the following command in your terminal:
++
+[source,terminal]
+----
+$ oc create -f <machine_config_file_name>
+----
+
+. Create an Open vSwitch (OVS) bridge, `br-ex1`, on nodes by using the OVN-Kubernetes network plugin to create an `extra_bridge` file`. Ensure that you save the file in the `/etc/ovnk/extra_bridge` path of the host. The file must state the interface name that supports the additional bridge and not the default interface that supports `br-ex`, which holds the primary IP address of the node. 
++
+.Example configuration for the `extra_bridge` file, `/etc/ovnk/extra_bridge`, that references a secondary interface that is attached to a VLAN
+[source,text]
+----
+bond1
+----
+
+. Create a machine configuration manifest file that defines the existing static interface that hosts `br-ex1` on any nodes restarted on your cluster:
++
+.Example of a machine configuration file that defines `bond1` as the interface for hosting `br-ex1`
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 12-worker-extra-bridge
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+        - path: /etc/ovnk/extra_bridge
+          mode: 0420
+          overwrite: true
+          contents:
+            source: data:text/plain;charset=utf-8,bond1
+          filesystem: root
+----
+
+. Apply the machine-configuration to your selected nodes:
+
+[source,terminal]
+----
+oc create -f <machine_config_file_name>
+----
+
+//Add to a separate file. 
+//. Bind `br-ex1` to the secondary interface by configuring the `/var/lib/ovnk/iface_default_hint` file. This file also specifies the interface for `br-ex`, which is the default `br-ex` bridge.
```

Comment:
Let's do include this after all (suggested):

~~~
. Optional: You can override/enforce br-ex selection logic by creating a machine-configuration file to create `/var/lib/ovnk/iface_default_hint` with the name of the iface you wish br-ex to select. By default, br-ex will select the primary interface of the node based on boot order and IP address subnet in the expected machine network. It may be necessary to ensure that br-ex continues to select the previously expected interfaces or bonds on your host depending on your configuration. 

[source,yaml]
----
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: worker
  name: 12-worker-br-ex-override
spec:
  config:
    ignition:
      version: 3.2.0
    storage:
      files:
        - path: /var/lib/ovnk/iface_default_hint
          mode: 0420
          overwrite: true
          contents:
            source: data:text/plain;charset=utf-8,bond0
          filesystem: root
----

. Apply the machine-configuration to your selected nodes:

[source,terminal]
----
oc create -f <machine_config_file_name>
----

[IMPORTANT]
====
Creation of this file should be used to enforce selection logic and should not be used to change the IP of existing nodes in the cluster. Changing node IP is not supported. Use this manifest during creation of a new node, or to ensure that br-ex is selecting the correct interface every time and does not conflict with newly created interfaces for use with `br-ex1`

It is advisable to manually create this file on a host node and reboot to test/confirm effects before committing this change to machine-config across all nodes.
====

~~~